Use anyhow for errors in logging

dfreese opened the pull request:
This is an example of what I brought up in #127.  I think it consolidates quite a bit of logic while also providing more of the original error back to the user.  There are pros and cons to that, if folks start to depend on that, but they may already start depending on the specific error string now, so it doesn't seem like a huge issue.

Putting this up also to run through CI, since I don't have a linux machine handy for running tests on at the moment.

lucab commented:
Thanks for the PR, but overall I'm inclined to reject this patch. I've left a longer explanation at https://github.com/lucab/libsystemd-rs/issues/127#issuecomment-1354424545, including a possible alternative approach.

dfreese commented:
I've updated this to remove anyhow and implement an internal `Context` that implements the expected formatting.  I've added a `SdResult` helper as well.  This is only implemented in logging.  Let me know if you'd rather it be adopted throughout the library more broadly in a single PR or have other concerns.

lucab commented on src/errors.rs:
```diff
@@ -32,3 +32,58 @@ pub(crate) enum ErrorKind {
 }
+
+/// Helper type for faliable functions within this library.
+pub type SdResult<T> = Result<T, SdError>;
```
I'd rather not introduce this, I don't see a real need for it.

dfreese replied:
removed

lucab commented:
Looks good to me, feel free to squash everything into a single commit and I'll merge it.
I'm fine to have it only applied to  the `logging` module for the moment, we can add the rest incrementally and new logic can already start using the context helpers.

dfreese commented:
Squashed.  Thanks for reviewing.
